pkg/server/plugin/nodeattestor/aws/iid.go

```diff
@@ -311,37 +321,37 @@ func (p *IIDAttestorPlugin) getConfig() (*IIDAttestorConfig, error) {
 	defer p.mtx.RUnlock()
 
 	if p.config == nil {
-		return nil, errors.New("not configured")
+		return nil, iidError.New("not configured")
 	}
-	return p.config, nil
+
+	config := *p.config
+	return &config, nil
```

Comment:
Why the copy of the struct? the config struct is replaced on each call to configure so this was safe...

Reply:
Yes, you are right, will fix it, thanks!